graphix/graphsim/rxgraphstate.py

```diff
@@ -89,10 +89,7 @@ def remove_node(self, node: int) -> None:
         nidx = self.nodes.get_node_index(node)
         self._graph.remove_node(nidx)
         self.nodes.remove_node(node)
-        edge_list = list(self.edges)
-        for e in edge_list:
-            if node in e:
-                self.edges.remove_edge(e)
+        self.edges.remove_edges_by_node(node)
 
```

Comment:
I think this part worsen the performance of `RXGraphState`, since it searches all the edges in the graph every time removing the nodes. This is fixed, so the performance of `RXGraphState` will be at least similar to that of `NXGraphState`.

Reply:
Great! Thanks!